doc/releases/changelog-dev.md

```diff
@@ -4,6 +4,29 @@
 
 <h3>New features since last release</h3>
 
+* Many parametrized operations now have the attribute `ndim_params` and
+  allow arguments with a broadcasting dimension. Also see entries for #2590
+  and #2575 below for details.
+  [(#2609)](https://github.com/PennyLaneAI/pennylane/pull/2609)
+
+  Previously unsupported broadcasted parameters are allowed for example in standard
+  rotation gates and matrix operations. The broadcasted dimension is the first dimension
+  in numerical operator representations. Note that the broadcasted parameter
+  has to be passed as an `array` but not as a python `list` or `tuple` for most operations.
```

Comment:
It might be worth re-phrasing this; the way it is currently worded, it sounds like there is some context missing 🤔 E.g., 'Previously unsupported broadcasted parameters...'

Perhaps the issue are?

- broadcasting in the context of operations and operation parameters has never been defined?
- Starting by saying what _is_ now allowed, rather than what wasn't previously allowed?